tests/unit/array/storage/elastic/test_del.py

```diff
@@ -0,0 +1,102 @@
+from unittest.mock import MagicMock
+
+import docarray.array.storage.elastic.backend as elastic_backend
+from docarray.array.storage.base.helper import Offset2ID
+
+MOCK_ES_OFFSET_INDEX = {
+    0: {"_id": 0, "_source": {"blob": "r0"}},
+    1: {"_id": 1, "_source": {"blob": "r1"}},
+    2: {"_id": 2, "_source": {"blob": "r2"}},
+    3: {"_id": 3, "_source": {"blob": "r3"}},
+    4: {"_id": 4, "_source": {"blob": "r4"}},
+    5: {"_id": 5, "_source": {"blob": "r5"}},
+    6: {"_id": 6, "_source": {"blob": "r6"}},
+    7: {"_id": 7, "_source": {"blob": "r7"}},
+    8: {"_id": 8, "_source": {"blob": "r8"}},
+}
+
+BUG_OFFSET_INDEX_AFTER_DELETION = {
+    0: {"_id": 0, "_source": {"blob": "r3"}},
+    1: {"_id": 1, "_source": {"blob": "r4"}},
+    2: {"_id": 2, "_source": {"blob": "r5"}},
+    3: {"_id": 3, "_source": {"blob": "r6"}},
+    4: {"_id": 4, "_source": {"blob": "r7"}},
+    5: {"_id": 5, "_source": {"blob": "r8"}},
+    6: {"_id": 6, "_source": {"blob": "r6"}},
+    7: {"_id": 7, "_source": {"blob": "r7"}},
+    8: {"_id": 8, "_source": {"blob": "r8"}},
+}
+
+EXPECTED_INDEX_AFTER_DELETION = {
+    0: {"_id": 0, "_source": {"blob": "r3"}},
+    1: {"_id": 1, "_source": {"blob": "r4"}},
+    2: {"_id": 2, "_source": {"blob": "r5"}},
+    3: {"_id": 3, "_source": {"blob": "r6"}},
+    4: {"_id": 4, "_source": {"blob": "r7"}},
+    5: {"_id": 5, "_source": {"blob": "r8"}},
+}
+
+ES_BULK_COUNT = 0
+
+
+class MockElasticIndices:
+    def __init__(self, **kwargs):
+        pass
+
+    def exists(self, **kwargs):
+        return True
+
+    def refresh(self, **kwargs):
+        pass
+
+
+class MockElasticClient:
+    def __init__(self, **kwargs):
+        self.indices = MockElasticIndices()
+
+    def count(self, **kwargs):
+        return {"count": len(MOCK_ES_OFFSET_INDEX)}
+
+
+def mock_bulk_index(*args, **kwargs):
+    global ES_BULK_COUNT
```

Comment:
is there a way to avoid `global` variables?

Reply:
okay will try to update this

Reply:
done